src/options.js

```diff
@@ -210,6 +207,38 @@ export function isNumeric(values) {
   }
 }
 
+export function isColors(values) {
+  for (const value of values) {
+    if (value == null) continue;
+    return isColor(value);
+  }
+}
+
+// Whereas isColors only tests the first defined value and returns undefined for
+// an empty array, this tests all defined values and only returns true if all of
+// them are valid colors. It also returns true for an empty array, and thus
+// should generally be used in conjunction with isColors.
+export function isAllColors(values) {
+  for (const value of values) {
+    if (value == null) continue;
+    if (!isColor(value)) return false;
+  }
+  return true;
+}
+
+// Mostly relies on d3-color, with a few extra color keywords. Currently this
+// strictly requires that the value be a string; we might want to apply string
+// coercion here, though note that d3-color instances would need to support
+// valueOf to work correctly with InternMap.
+export function isColor(value) {
+  if (!(typeof value === "string")) return false;
+  value = value.toLowerCase();
```

Comment:
that's new. css colors are case-insensitive, so should we mention it as a bugfix?

Reply:
Yes, this does fix a bug. I’d be surprised if anyone encountered it, but we could mention it.